sql/core/src/main/scala/org/apache/spark/sql/execution/subquery.scala

```diff
@@ -194,6 +194,11 @@ case class PlanSubqueries(sparkSession: SparkSession) extends Rule[SparkPlan] {
         }
         val executedPlan = new QueryExecution(sparkSession, query).executedPlan
         InSubqueryExec(expr, SubqueryExec(s"subquery#${exprId.id}", executedPlan), exprId)
+      case exists: expressions.Exists =>
+        val executedPlan = new QueryExecution(sparkSession, exists.plan).executedPlan
```

Comment:
can you update your local branch? AFAIK the latest code is using ` QueryExecution.prepareExecutedPlan` now.

Reply:
And I think we should create a new rule in the `Finish Analysis` batch to do the conversion, so that it can ho through the optimizer.

Reply:
> And I think we should create a new rule in the `Finish Analysis` batch to do the conversion, so that it can ho through the optimizer.

ok

Reply:
> Finish Analysis

How about current change. 

Add class `ReplaceNonCorrelatedExists ` comment later